Sources/SwiftKafka/KafkaAcknowledgedMessage.swift

```
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import NIOCore
+
+/// A message produced by the client and acknowledged by the Kafka cluster.
+public struct KafkaAcknowledgedMessage: Hashable {
+    /// The unique identifier assigned by the ``KafkaProducer`` when the message was send to Kafka.
+    /// The same identifier is returned by ``KafkaProducer/sendAsync(message:)`` and can be used to correlate
+    /// a sent message and an acknowledged message.
+    public let id: UInt?
+    /// The topic that the message was sent to.
+    public let topic: String
+    /// The partition that the message was sent to.
+    public let partition: Int32
+    /// The key of the message.
+    public let key: ByteBuffer?
+    /// The body of the message.
+    public let value: ByteBuffer
+    /// The offset of the message in its partition.
+    public let offset: Int64
+
+    /// Initialize `KafkaAckedMessage` from `rd_kafka_message_t` pointer.
+    init?(messagePointer: UnsafePointer<rd_kafka_message_t>, id: UInt? = nil) {
+        self.id = id
+
+        let rdKafkaMessage = messagePointer.pointee
+        guard rdKafkaMessage.err.rawValue == 0 else {
+            return nil
```

Can we actually throw the Kafka error here?